hdfs-dataset-sink/pom.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>hdfs-dataset-sink</artifactId>
+	<packaging>jar</packaging>
+	<name>hdfs-dataset-sink</name>
+	<description>HDFS Dataset Sink stream module</description>
+
+	<parent>
+		<groupId>org.springframework.cloud.stream.module</groupId>
+		<artifactId>spring-cloud-stream-modules</artifactId>
+		<version>1.0.0.BUILD-SNAPSHOT</version>
+	</parent>
+
+	<properties>
+		<start-class>org.springframework.cloud.stream.module.hdfs.dataset.DatasetSinkApplication</start-class>
+		<spring-data-hadoop.version>2.3.0.RC1</spring-data-hadoop.version>
+	</properties>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-hadoop-boot</artifactId>
+			<version>${spring-data-hadoop.version}</version>
+			<scope>provided</scope>
```

Comment:
Hmm, not sure I understand exactly how distro switching would work in your opinion.
Do you expect to create BOMs that transitively include `spring-data-hadoop-boot` and `spring-data-hadoop-store` for distros? If so, and several distros depend on different versions of the above, then the code for this module may fail at runtime anyway (given we would provide a different version of say, `org.springframework.data.hadoop.store.dataset.DatasetDefinition` from what was used at compile time)